csa_rescale_stat.py

```diff
@@ -46,175 +50,275 @@ def get_parser():
+    :param mean_CSA: mean value of CSA from which percentage atrophy is calculated (default = 80)
+    '''
+    fig6, ax = plt.subplots()
     # data for plotting
     n=[]
     for z_p in z_power:
-        i = np.arange(1.5, 8.0, 0.05)
-        i_perc = np.arange(1.5, 8.0, 0.05)
-        num_n = ((z+z_p)**2)*((2*std)**2)
+        i = np.arange(1.5, 8.0, 0.05) # x_axis values ranging from 1.5 to 8.0 mm^2
+        num_n = 2*((z+z_p)**2)*(std**2) # numerator of sample size equation
         n.append(num_n/((i)**2))
         # plot
     ax.plot(i, n[0], label=('80% power'))
     ax.plot(i, n[1], label=('90% power'))
-    ax.set_ylabel('minimum number of participants')
+    ax.set_ylabel('minimum number of participants per arm')
     ax.set_xlabel('atrophy in mm^2')
-    ax.set_title('minimum number of subjects to detect an atrophy ')
+    ax.set_title('minimum number of subjects to detect an atrophy with 5% uncertainty\n std = '+str(round(std,2))+'mm², mean_CSA = '+str(mean_CSA)+'mm²')
     ax.legend()
     ax.grid()
-    # TODO: change: this functions should take into account variable mean_CSA, default=80
+    # create global variable for secax functions
+    global mean_CSA_sample
+    mean_CSA_sample = mean_CSA
+
     def forward(i):
-        i2 = i/80*100
+        i2 = i/mean_CSA_sample*100
         return i2
     def inverse(i):
-        return i/100*80
+        return i/100*mean_CSA_sample
     secax = ax.secondary_xaxis('top', functions=(forward, inverse))
     secax.set_xlabel('atrophy in %')
-    fig.savefig("min_subj.jpg", bbox_inches='tight')
+    plt.savefig("min_subj.jpg", bbox_inches='tight')
 
 
+def std(df_a, Vertlevels):
+    '''
+    Computes standard deviation of subject mean CSA for each rescaling and different vertebrae levels
+    '''
+    print("\n====================std==========================\n")
+    # TODO: normalization of CSA for intersubject studies
+    min_vert = min(list(Vertlevels))
+    for i in Vertlevels[1:]:
+        for name, group in df_a.groupby('Rescale'):
+            gt_CSA = 'CSA_C'+str(min_vert)+'_C'+str(i)
+            std = group[gt_CSA].std()
+            atrophy = set(group.reset_index().Rescale)
+            print('CSA std on '+str(atrophy)+'  rescaled image C'+str(min_vert)+'/C'+str(i)+' is ',round(std, 3),' mm^2 ')
+        print('\n')
+
 
-# Main
-########################################################################
-def main():
+def sample_size(df_a, conf, power, mean_control, mean_patient):
+    '''
+    calculate the minimum number of patients required to detect an atrophy of X (i.e. power analysis)
+    sample size with certainty 95% z(0.05/2)=1.96, power 80% z(Power = 80%)=0.84, ratio patients/control 1:1
+    and with the assumption both samples have same std
+    :param df: dataframe containing csv files values
+    :param conf: Confidence level (i.e. 0.60, 0.70, 0.8, 0.85, 0.90, 0.95)
+    :param power: Power level (i.e. 0.60, 0.70, 0.8, 0.85, 0.90, 0.95)
```

Comment:
no need to give all those values, that's confusing
```suggestion
    :param power: Power level. Example: 0.9
```